app/inpututils.cpp

```diff
@@ -1354,15 +1381,65 @@ QString InputUtils::distanceToFeature( QgsPoint gpsPosition, const FeatureLayerP
     catch ( QgsCsException &e )
     {
       Q_UNUSED( e )
-      return QString();
+      return -1;
     }
   }
 
   QgsDistanceArea distanceArea;
   distanceArea.setSourceCrs( mapSettings->destinationCrs(), mapSettings->transformContext() );
-  double dist = distanceArea.measureLine( gpsPosition, targetPoint );
-  QString res = formatDistance( dist, distanceArea.lengthUnits(), 2 );
-  return res;
+  qreal distance = distanceArea.measureLine( gpsPosition, targetPoint );
+  distance = distanceArea.convertLengthMeasurement( distance, QgsUnitTypes::DistanceMeters );
+  return distance;
+}
+
+qreal InputUtils::angleBetweenGpsAndFeature( QgsPoint gpsPoint, const FeatureLayerPair &targetFeature, QgsQuickMapSettings *mapSettings )
+{
+  if ( !mapSettings || !targetFeature.isValid() )
+    return -1;
+
+  QgsVectorLayer *layer = targetFeature.layer();
+  QgsFeature f = targetFeature.feature();
+
+  // Only points are supported
+  if ( layer->geometryType() != QgsWkbTypes::GeometryType::PointGeometry )
+    return -1;
+
+  // Transform gps position to map CRS
+  QgsCoordinateReferenceSystem wgs84( "EPSG:4326" );
+  QgsCoordinateTransform ct1( wgs84, mapSettings->destinationCrs(), mapSettings->transformContext() );
+  if ( !ct1.isShortCircuited() )
+  {
+    try
+    {
+      gpsPoint.transform( ct1 );
+    }
+    catch ( QgsCsException &e )
+    {
+      Q_UNUSED( e )
+      return -1;
+    }
+  }
+
+  // Transform target point to map CRS
+  QgsPoint target( extractPointFromFeature( targetFeature ) );
+  QgsCoordinateTransform ct2( targetFeature.layer()->crs(), mapSettings->destinationCrs(), mapSettings->transformContext() );
```

Comment:
maybe wrap in some util function? (the same as above)

Reply:
Can you please check again if it is fine now? I changed it in this commit: https://github.com/lutraconsulting/input/pull/1920/commits/76d987e781be54c2c4cb81873b74b980f2a0ac00

Reply:
it is ok